app/src/main/java/com/jetbrains/handson/mpp/mobile/TrainDataAdapter.kt

```diff
@@ -1,35 +1,66 @@
 package com.jetbrains.handson.mpp.mobile
 
 import android.annotation.SuppressLint
+import android.icu.util.TimeUnit
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import kotlinx.android.synthetic.main.list_row.view.*
+import java.text.DateFormat
 import java.text.SimpleDateFormat
+import java.time.Instant
+import java.time.LocalDate
+import java.time.Period
+import java.time.format.DateTimeFormatter
 import java.util.*
 
 class TrainDataAdapter(private val dataSet: List<TrainTimes.Journey>) :
     RecyclerView.Adapter<TrainDataAdapter.MyViewHolder>() {
 
     // Provide a reference to the views for each data item
     class MyViewHolder(private var view: View) : RecyclerView.ViewHolder(view) {
+        private fun getTimeString(hour: Int, min: Int): String {
+            return "%02d:%02d".format(hour, min)
+        }
+
         @SuppressLint("SetTextI18n")
         fun addJourney(j: TrainTimes.Journey) {
+            /*
+            Not as clean as it could be due to the API level 23.
+             */
             view.changes.text = j.numberChanges.toString()
             view.price.text = "£${j.price/100}.${j.price % 100}"
```

Comment:
what happens if `j.price % 100 < 10`? Do we get `£15.5` rather than `£15.05`?

Reply:
Good catch.